Make various functions more efficient.

luciansmith opened the pull request:
In particular:
* The colors list was being constantly copied everywhere.  Now it's never copied at all.
* Any time a set position or size function was called, the curves were always updated a la autolayout.  It's still possible to move things with that functionality, but by default, it no longer does this.

adelhpour reviewed (commented):
- we still need to call 'roundToTwoDecimalPlaces' when getting the position and dimensions values of layout elements. I can merge this one and then add them myself (but I want you to leave a 'todo' or something so that I remember adding them.
- 'c_api_getNumValidColorNameValues', 'c_api_getNthValidColorNameValue', 'c_api_getNumValidHexColorCodeValues', and 'c_api_getNthValidHexColorCodeValue' are used to get a list of available colors on Python side. Again, leaving a todo or something is helpful here.
- I know it doesn't sound intuitive, but I feel like setting the default value of 'updateCurves' to 'true' is more useful, because the user usually wants to move the nodes and see how the curves are aligned to its new position accordingly (but still not sure which one works best). Also, I think  we need to remove the definition of 'canUpdateLayoutCurves' from the code as it is no longer used anywhere.

luciansmith commented:
OK, I addressed your requests!

* Added new issue here for putting back in the 'round' function.  However, I can say that the old system wasn't actually working as advertised; I always got non-rounded versions of almost every value even before this PR.
* Changed the default updateCurves value to true.
* Put in a new design in the Python bindings that just uses the 'colorData' function.  It might not work yet.  But once 'colorData' returns a python dictionary (which it should be able to do) everything else will work.  Also, I renamed the functions, since they weren't always returning lists, and they weren't returning 'all valid' values, but rather the predefined values.  Unpredefined hex values are still legal.

adelhpour reviewed (approved):
All sounds good to me!
